### DISCUSSION TITLE
The case for animations / transitions

### DISCUSSION THREAD

**Discussion by poirazis** [Budibase/budibase#7510] [Ideas]
The Conditional UI capabilities, give us the power to show / hide / resize etc Widgets at runtime.
Do we need animations / transitions, and how would we get them ?

Lets take a simple example.
We have a 4 step form. Each time the Next button is pressed we go from one step to the next.
Now, the Steps have no animation, they just disappear and appear, rudely :)
In the modern world, and the level of UX / UI , it would be nicer if these transitions were smoothed.

So, how do we get this effect ? 

The hackish way : 
we can add transition effects using an Embed, and targeting the elements we want to add transitions to.
Example here : #7511 

The proposed way : 

    Conditional UI : we can add a transition property as an optional parameter to the Conditional UI.

    Example 1 : [ Show Component IF State.detailsExpanded === true with TRANSiTiON fadeIn, 600ms ease-in-out ]
    This rule would set the proper CSS so the Widget animates its transition from one State to Another.

    Example 2 : [ Update Setting / Color = green IF Data Provider.Rows is not Empty with TRANSITION crossFade 200ms ]

    The Widgets / Screens : we can add a default Into / Outro transition to every widget ( assigned to the wrapper div) that would enable to set the transition of how the widget appears on screen. 
    Example 1 : Title widget : Default Intro : Typewriter / 500ms / ease-in
                                               Default Outro :  FadeOut / 200ms / ease

Just an idea!

**Comment by R2bEEaton** [comment]
Love it!

**Reply by melohagan** [reply]
Cool idea! There's an issue open that looks related here: https://github.com/Budibase/budibase/issues/3712

**Comment by shogunpurple** [comment]
Hi @poirazis 

We actually used to have this - using svelte transitions, but opted to remove it as it wasn't popular. If this is something people request and would find useful, we could look at adding support again.

Thanks for the suggestion!
